### PR TITLE
Lde/honk transcript updates

### LINKED ISSUE #16: Integrate transcript into Sumcheck
- add methods for serialization/deserialization of univariate to transcript and corresponding tests
- integrate existing transcript into Sumcheck

### PR DESCRIPTION
**Pull request by ledwards2225**
- Add some basic transcript functionality into the HonkProver skeleton, e.g. adding data to transcript, computing round challenges
- Clean up some related comments in the HonkProver
- Add a test for serialization/deserialization of a univariate to/from a transcript; needed for Sumcheck

### CODE REVIEW

File: cpp/src/aztec/honk/sumcheck/polynomials/univariate.hpp [hunk 1]
```diff
@@ -38,8 +43,39 @@ template <class Fr, size_t _length> class Univariate {
         }
     }
 
+    // TODO(luke): I'd like to be able to construct a Univariate directly from a buffer this leads to ambiguity with the
```
Comment (codygunton) [inline cpp/src/aztec/honk/sumcheck/polynomials/univariate.hpp]: This is resolved by `serialize_from_buffer` defined below?

File: cpp/src/aztec/honk/sumcheck/polynomials/univariate.hpp [hunk 2]
```diff
@@ -38,8 +43,39 @@ template <class Fr, size_t _length> class Univariate {
...
+    //     ASSERT(num_elements == _length);
+    //     for (size_t i = 0; i < num_elements; ++i) {
+    //         evaluations[i] = from_buffer<Fr>(buffer, i * sizeof(Fr));
+    //     }
+    // }
+
     Fr& value_at(size_t i) { return evaluations[i]; };
 
+    // Write the Univariate evaluations to a buffer
+    std::vector<uint8_t> to_buffer() const
```
Comment (codygunton) [inline cpp/src/aztec/honk/sumcheck/polynomials/univariate.hpp:46]: Did you try to implement in terms of the prescription [here](https://github.com/AztecProtocol/barretenberg/blob/ab331f74c3a00ae0babd531ecb3d300365aae483/cpp/src/aztec/common/serialize.hpp#L9) and run into trouble?

File: cpp/src/aztec/honk/sumcheck/polynomials/univariate.hpp [hunk 3]
```diff
@@ -213,7 +249,7 @@ template <class Fr, size_t view_length> class UnivariateView {
     Fr& value_at(size_t i) { return evaluations[i]; };
 
     template <size_t full_length>
-    explicit UnivariateView(Univariate<Fr, full_length> univariate_in)
+    explicit UnivariateView(Univariate<Fr, full_length>& univariate_in)
```
Comment (codygunton) [inline cpp/src/aztec/honk/sumcheck/polynomials/univariate.hpp:235]: Crucial!

File: cpp/src/aztec/honk/sumcheck/sumcheck.hpp [hunk 4]
```diff
@@ -39,24 +45,28 @@ template <class Multivariates, class ChallengeContainer, template <class> class.
...
+        auto round_univariate = round.compute_univariate(multivariates.full_polynomials, relation_separator_challenge);
+        transcript.add_element("univariate_0", round_univariate.to_buffer());
         // IMPROVEMENT(Cody): Could move building of this list into challenge container?
-        round_challenges[0] = challenges.get_sumcheck_round_challenge(0);
+        round_challenges[0] = transcript.get_mock_challenge();
         multivariates.fold(multivariates.full_polynomials, multivariate_n, round_challenges[0]);
 
         // All but final round
-        // We operate on multivariates.full_polynomials in place.
+        // We operate on multivariates.folded_polynomials in place.
```
Comment (codygunton) [inline cpp/src/aztec/honk/sumcheck/sumcheck.hpp:56]: ...indeed :facepalm:

File: cpp/src/aztec/honk/sumcheck/sumcheck.test.cpp [hunk 5]
```diff
@@ -59,43 +53,45 @@ TEST(Sumcheck, Prover)
...
-                             ArithmeticRelation,
-                             GrandProductComputationRelation,
-                             GrandProductInitializationRelation>(challenges);
-
+    // TODO(luke): verifier now requires prover to have executed; executing verifier here for now.
     sumcheck.execute_verifier();
-    // TODO(Cody) This does not constitute a test.
 }
 
+// TODO(luke): Fails because verifier now requires prover to have run; execute_verifier() run in previous test
```
Comment (codygunton) [inline cpp/src/aztec/honk/sumcheck/sumcheck.test.cpp]: Why?

File: cpp/src/aztec/honk/sumcheck/polynomials/univariate.hpp (hunk 2 above)
Comment (codygunton) [reply cpp/src/aztec/honk/sumcheck/polynomials/univariate.hpp:46]: ...oh, we put stuff in the `std` namespace?? TODO: are we violating https://en.cppreference.com/w/cpp/language/extending_std?

File: cpp/src/aztec/honk/sumcheck/sumcheck.test.cpp (hunk 5 above)
Comment (codygunton) [reply cpp/src/aztec/honk/sumcheck/sumcheck.test.cpp]: TODO: mock this and make it a real unit test.

**Review by codygunton — CHANGES_REQUESTED** [review]
Please reinstate the verifier test with mocking. Good otherwise.

File: cpp/src/aztec/honk/sumcheck/polynomials/univariate.hpp (hunk 1 above)
Comment (ledwards2225) [reply cpp/src/aztec/honk/sumcheck/polynomials/univariate.hpp]: Yeah, removed this commented out constructor/comment

File: cpp/src/aztec/honk/sumcheck/sumcheck.test.cpp (hunk 5 above)
Comment (ledwards2225) [reply cpp/src/aztec/honk/sumcheck/sumcheck.test.cpp]: Done

**Review by codygunton — APPROVED** [review]

**Pull request merged**